### PR TITLE
doc: net: Enhance DNS resolver documentation

### PR DESCRIPTION
**Pull request by jukkar**
Add more information about DNS resolver to documentation. Move
generic DNS information from the sample application into DNS
resolver documentation.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

File: doc/reference/networking/dns_resolve.rst [hunk 1]
```diff
@@ -6,6 +6,30 @@ DNS Resolve
 Overview
 ********
 
+The DNS resolver sample application implements a basic DNS resolver according
+to https://tools.ietf.org/html/rfc1035.
```
Comment (dbkinder) [inline doc/reference/networking/dns_resolve.rst]:
```suggestion
to `IETF RFC1035 on Domain Implementation and Specification <https://tools.ietf.org/html/rfc1035>`_.
```

File: doc/reference/networking/dns_resolve.rst [hunk 2]
```diff
@@ -6,6 +6,30 @@ DNS Resolve
...
+Supported DNS answers are IPv4/IPv6 addresses and CNAME.
+
+If a CNAME is received, the DNS resolver will create another DNS query.
+The number of additional queries is controlled by the
+DNS_RESOLVER_ADDITIONAL_QUERIES Kconfig variable.
```
Comment (dbkinder) [inline doc/reference/networking/dns_resolve.rst]:
```suggestion
option:`CONFIG_DNS_RESOLVER_ADDITIONAL_QUERIES` Kconfig variable.
```

File: doc/reference/networking/dns_resolve.rst [hunk 3]
```diff
@@ -6,6 +6,30 @@ DNS Resolve
...
+
+The multicast DNS (mDNS) client resolver support can be enabled by setting
+:option:`CONFIG_MDNS_RESOLVER` Kconfig option.
+See https://tools.ietf.org/html/rfc6762 for more details about mDNS.
```
Comment (dbkinder) [inline doc/reference/networking/dns_resolve.rst]:
```suggestion
See `IETF RFC6762 <https://tools.ietf.org/html/rfc6762>`_ for more details about mDNS.
```

File: doc/reference/networking/dns_resolve.rst [hunk 4]
```diff
@@ -6,6 +6,30 @@ DNS Resolve
...
+
+The link-local multicast name resolution (LLMNR) client resolver support can be
+enabled by setting the :option:`CONFIG_LLMNR_RESOLVER` Kconfig option.
+See https://tools.ietf.org/html/rfc4795 for more details about LLMNR.
```
Comment (dbkinder) [inline doc/reference/networking/dns_resolve.rst]:
```suggestion
See `IETF RFC4795 <https://tools.ietf.org/html/rfc4795>`_ for more details about LLMNR.
```

**Review by dbkinder — DISMISSED** [review]
looks good, with a few edits...

**Comment by jukkar** [comment]
Updated according to comments.

**Review by nashif — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
looks good, thanks!

**Pull request merged**